src/dataset.py

```diff
@@ -1,24 +1,94 @@
+import csv
+
 import numpy as np
-from typing import Dict
+from typing import List, Optional, Tuple
+from loss import Information_gain
 
 
 class Dataset:
-    def __init__(self, dataset: Dict[str, np.ndarray]):
+    def __init__(self, dataset: np.array, labels_index: int):
         self._data = dataset
+        self._labels_index = labels_index
 
-    def load_from_csv(self, path) -> 'Dataset':
-        pass
+    # Idea for the loading data: creating another module which allows to generalize the format of the data imported
+    # Using for example an abstract class for the loading package.
 
-    def load_from_array(self, data: np.array) -> 'Dataset':
-        pass
+    @staticmethod
+    def load_from_csv(path: str, csv_limiter: str, label_name: str, new_line: str = '') -> 'Dataset':
 
+        with open(path, newline=new_line) as csvfile:
+            data = []
+            reader = csv.reader(csvfile, delimiter=csv_limiter)
+            for row in reader:
+                data.append(row)
 
-    def chose_best_split(self):
-        # TODO : This function contains the algo which allows us to take the split from the dataset. I have to think about the link of a condition
-        pass
+            label_index = data[0].index(label_name)
+            data = np.asarray(data)
+            data = data[1:]
 
-    def split_condition(self, condition):
-        # TODO : This function allows us to split the dataset given a condition. The format of the conditon is still unknown
-        pass
+        dataset = Dataset(dataset=data, labels_index=label_index)
+        return dataset
+
+    @staticmethod
+    def load_from_array(data: np.array, label_index: int) -> 'Dataset':
+        dataset = Dataset(dataset=data, labels_index=label_index)
+        return dataset
+
+    def _get_possible_split(self):
+        all_possibilites = {}
+
+        for i in range(self._data.shape[0]):
+            if i == self._labels_index:
+                continue
+
+            all_possibilites[i] = np.unique(self._data[:, i])
+        return all_possibilites
+
+    def _best_split_for_each_feature(self, purity_evaluation: Information_gain):
+        ig_gain_by_feature = {}
+        possibilites = self._get_possible_split()
```

Comment:
a bit pedantic but maybe I'd change possibilities to something like split_value_possibilities just to make it easier to follow later on

Reply:
Yes

Reply:
Although possible_split_values is better